### ISSUE TITLE
Seatmap

### ISSUE THREAD

**Issue by agreements** [MetalMichael/LAN-Website#2]
Just wondering how i can change this, and the setup, also making it work.

**Comment by silverforest** [comment]
Yeah, I see how it works, but there has to be some system/program for setting it up

**Comment by MetalMichael** [comment]
I think I saw on someone's site that they'd figured out how to change how the map looks, so that's not a problem, I won't cover that.

One important thing that is completely missing from everything included here, is an authentication system we have at our LAN events. This runs on a separate LAN server we have. Before anyone can access the internet through LAN, they have to authenticate with this server. Users enter their account details and seat number. This details are authenticated and then seat saved. Seat number is tied to a ticket. IP addresses are then imported to the router's auth table.

While all of this is happening, a background php script is running. See: daemons/map-daemon.php
This runs once every X seconds (defined in settings), asynchronously. And updates the info on the map.

There's also a small bash script that runs once/minute (minimum cron interval) that ensures all of the daemons are running. Something like this:

```
#!/bin/sh -x
#Map daemon
ls /var/run/screen/S-soc_lsucs/*.lanwebsite-map-daemon >/dev/null 2>&1 || screen -A -m -d -S lanwebsite-map-daemon php5 /home/soc_lsucs/lan.lsucs.org.uk/htdocs/daemons/map-daemon.php

#Chat daemon
ls /var/run/screen/S-soc_lsucs/*.lanwebsite-chat-daemon >/dev/null 2>&1 || screen -A -m -d -S lanwebsite-chat-daemon php5 /home/soc_lsucs/lan.lsucs.org.uk/htdocs/daemons/chat-daemon.php

#Lobby daemon
ls /var/run/screen/S-soc_lsucs/*.lanwebsite-lobby-daemon >/dev/null 2>&1 || screen -A -m -d -S lanwebsite-lobby-daemon php5 /home/soc_lsucs/lan.lsucs.org.uk/htdocs/daemons/lobby-daemon.php
```

Most important bit missing seems to be the way of getting users to enter their seat numbers.
